code/modules/mob/living/living_defense.dm

```diff
@@ -422,13 +422,16 @@
 		bodytemperature += round(BODYTEMP_HEATING_MAX*(1-thermal_protection), 1)
 		if(world.time >= next_onfire_brn)
 			next_onfire_brn = world.time + 50
```

Comment:
```suggestion
			next_onfire_brn = world.time + 50
			if(fire_stacks <= 0)
				ExtinguishMob()
			if(fire_stacks > FIRE_MAX_STACKS)
				fire_stacks = FIRE_MAX_STACKS //Hardcap to prevent gamers from applying 300 firestacks to a mob or player. That way people dont burn for all eternity.
```